crates/nu-std/std/log.nu

```diff
@@ -135,129 +142,158 @@ def now [] {
 }
 
 # Log an info message
 export def info [
     message: string, # A message
     --short (-s) # Whether to use a short prefix
+    --format (-f): string # A format (for further reference: help std log)
 ] {
-    if (current-log-level) > (INFO_LEVEL) {
-        return
-    }
-
-    let prefix = if $short {
-        INFO_LEVEL_PREFIX --short
-    } else {
-        INFO_LEVEL_PREFIX
-    }
-    log-formatted (ansi default) $prefix $message
+    handle-log $message (log-types | get INFO) $format $short
 }
 
 # Log a debug message
 export def debug [
     message: string, # A message
     --short (-s) # Whether to use a short prefix
+    --format (-f): string # A format (for further reference: help std log)
 ] {
-    if (current-log-level) > (DEBUG_LEVEL) {
-        return
-    }
-
-    let prefix = if $short {
-        DEBUG_LEVEL_PREFIX --short
-    } else {
-        DEBUG_LEVEL_PREFIX
-    }
-    log-formatted (ansi default_dimmed) $prefix $message
+    handle-log $message (log-types | get DEBUG) $format $short
 }
 
-# Log a message with a specific format and verbosity level
-# 
-# Format reference:
-# - %MSG% will be replaced by $message
-# - %DATE% will be replaced by the timestamp of log in standard Nushell's log format: "%Y-%m-%dT%H:%M:%S%.3f"
-# - %LEVEL% will be replaced by the standard Nushell's log verbosity prefixes, e.g. "CRT"
-#
-# Examples:
-# - std log custom "my message" $"(ansi yellow)[%LEVEL%]MY MESSAGE: %MSG% [%DATE%](ansi reset)" (std log WARNING_LEVEL)
+# Log a message with a specific format and verbosity level, with either configurable or auto-deduced %LEVEL% and %ANSI_START% placeholder extensions
 export def custom [
     message: string, # A message
-    format: string, # A format
-    log_level: int # A log level
-    --short (-s) # Whether to use a short prefix
+    format: string, # A format (for further reference: help std log)
+    log_level: int # A log level (has to be one of the $env.LOG_LEVEL values for correct ansi/prefix deduction)
+    --level-prefix (-p): string # %LEVEL% placeholder extension
+    --ansi (-a): string # %ANSI_START% placeholder extension
 ] {
     if (current-log-level) > ($log_level) {
         return
     }
 
-    let level = ((if $short {
-        parse-int-level $log_level --short
-    } else {
+    let valid_levels_for_defaulting = [
+        $env.LOG_LEVEL.CRITICAL
+        $env.LOG_LEVEL.ERROR
+        $env.LOG_LEVEL.WARNING
+        $env.LOG_LEVEL.INFO
+        $env.LOG_LEVEL.DEBUG
+    ]
+
+    let prefix = if ($level_prefix | is-empty) {
+        if ($log_level not-in $valid_levels_for_defaulting) {
+            let span = (metadata $log_level).span
+            error make {
+                "msg": $"Cannot deduce level prefix for given log level: ($log_level)"
+                label: {
+                    text: "Invalid log level for prefix auto-deduction"
+                    start: $span.start
+                    end: $span.end
+                }
+            }
```

Comment:
i do not understand the error thrown by this :thinking: 

okey, it's because for instance in 
```bash
std log custom "msg" "%MSG%" 25
```
`25` is not a "round" log level, but i think it's confusing for the user :thinking: 

i'd say either
- allow intermediate values such as 25 by rounding
- give a more helpful error with the possible values

Reply:
i've refactored it a tiny bit in c3ad10f094a1f9bf06ab44cfca7f58842d635ff6 :yum: 

but yeah, listing the possible values looks really better to me, thanks :+1: